### ISSUE TITLE
Feature-Request: Pop-Out SQL Window

### ISSUE THREAD

**Issue by Hansjakobli123** [phpmyadmin/phpmyadmin#11983]
In earlier versions of phpmyadmin it was possible to pop out the SQL window and exectute commands in it. As I found out this feature was replaced by the Console Window at the bottom.

When using multiple screens it would be nice the old windows was more practical as you could move it to another screen. Opening a whole new window takes too much space however.

I don't really see why the two features are mutually exclusive, would it be possible to bring the popup window back or (probably even better) have a button to open the console in a new window?

**Comment by nijel** [comment]
These features are not mutually exclusive, however most people do prefer no separate windows these days, that's why it was replaced by console. Not sure how much used it would be. Maybe we could come with way to detach console from the main window...

**Comment by Hansjakobli123** [comment]
We are a team of 5 people using it, and 4/5 complain about not having this feature anymore since I upgraded phpmyadmin. They are nagging me I should downgrade to 3.9, but I don't really want to do that.

 I actually really like the new console, so if there was a way to detach it, that would be awesome!

**Comment by Achilles-96** [comment]
What do you mean by popping out SQL window. Can you point me to version where this feature is present?

**Comment by zixtor** [comment]
Version 4.2 should have it seems, please do git checkout tags/RELEASE_4_2_0 and look in navigation panel's top area for a query window.

**Comment by nijel** [comment]
Separate browser window where you could enter SQL query.

**Comment by Hansjakobli123** [comment]
I just spoke with my people and it seems the feature is mostly missed because if you create very long (screen filling) SQLs you can enter and tweak them on a another screen and you always see the results on your main screen without scrolling all the time. That's why the old query window was quite good for that.

**Comment by shev72** [comment]
I also miss this feature. I can no longer easily test queries quickly and see results in the other window.  This was very useful. Agree that pop up command would be equivalent.

**Comment by Achilles-96** [comment]
@nijel ,I am working on this feature.

**Comment by nategasser** [comment]
Another vote for bringing back this feature -- at least as an option.  Thanks!

**Comment by Achilles-96** [comment]
In the earlier version, a new window opens up whenever we submit a command. Are we having the same functionality or any different way of doing this. I mean instead of opening a new window, why can't we execute the command in same popped out window.

**Comment by Hansjakobli123** [comment]
Executing it in another window has the advantage of building and optimizing queries in on one screen and see the result on the other without scrolling.

**Comment by userdude** [comment]
@Achilles-96 From what I understand, there's a commit that will bring back (through a configuration setting I presume) the query window (which for me is really, really useful; I write long queries over several hours, and oftentimes need to review the output as I'm editing). We're on 4.6.6; do you know if/when this feature will come back?

Also, an aside, since the new console/postback edit box took over, comments are being stripped out, which sucks because the methodology requires you to submit it to see the results. Is this related, or should I open a separate issue?

**Comment by msykes** [comment]
this was uber useful, it's harder to work on longer queries this way. looking forward to this being brought back!

**Comment by Woodgnome** [comment]
Recently started using the newer versions of phpMyAdmin - having a hard time describing how much the console pisses me off compared to the standalone window...

**Comment by kmchen13** [comment]
Same request here. I consider to downgrade phpmyadmin on Debian 9

**Comment by ojde** [comment]
Same request here. With complex SQL queries this was the very best way for developing and I sorely miss it.

**Comment by yurtkuran** [comment]
```
while (!PopOutSQLWindow) {
    echo "Please bring back Pop-Out SQL Window";
}
```

**Comment by fengyqf** [comment]
Same request here. 
At least, bring back it as an option, whitch not enabled default.

**Comment by freddythunder** [comment]
It's been 1,187 days since this post began.  I'm thinking no one cares that people like us enjoyed the pop out SQL window.  The up arrow doesn't seem to work correctly on any versions of phpmyadmin that I use at home nor at work.  I'd like it back too please.

**Comment by drakgoku** [comment]
I am a strong advocate for phpmyadmin, but sometimes they don't do things correctly.

If the creators themselves are not interested, we will always have workbench (that section does it very well, having multiple sheets and on top of that they are saved)

**Comment by williamdes** [comment]
Hi everybody commenting
I am very sorry that this issue looks like it is ignored, we really want to listen to our users so phpMyAdmin could fit best their needs

Sometimes issues need a little bump to get more attention :)

I am assigning this to the milestone 5.2 in order to get this done some time soon

**Comment by drakgoku** [comment]
It is not to hurry anyone, since this was published on "18 Feb 2016" and we are at "02 AUG 2021" that is more than 5 years and 6 months, with a simple "pop-up". We are not asking you for "magic" like current editors or intellij do.

Perfectly, we could ask for a sheet manager and a display manager or simply switch to MySQL Workbench, if not use other software. Faced with incompetence, you have to go for competition.

I understand that the "supposed community manager, if there was one" of phpMyAdmin has not programmed in its life in any editor, it is understandable, but the programmers who program "a lot", have a window where you execute the code elsewhere, that is essential.

I endorse that there are many good programs, Microsoft SQL Server, MySQL Workbench, HeidiSQL, visual code ... with many colors and many nonsense that make your work easy.
I could list the top of 2021, but it would be too much competition for you.

Putting a simple pop-up is modifying JavaScript. It costs them nothing or it may be that phpMyAdmin "workers / creators" use other tools, that would be more understood.

Anything free is expensive.

How could AMD beat INTEL? Life is very simple. But there are those who do not see it, this was seen to come from 2006 since the purchase of ATI.

**Comment by williamdes** [comment]
Some references for someone working on this:
- https://github.com/phpmyadmin/phpmyadmin/blob/RELEASE_4_2_11/doc/config.rst had `EditInWindow` config option
- https://github.com/phpmyadmin/phpmyadmin/blob/RELEASE_4_2_11/libraries/config.default.php#L2518-L2524 is the default config block

So `EditInWindow` should be added back and valued default as `false`

```rst

.. config:option:: $cfg['EditInWindow']

    :type: boolean
    :default: true
```

**Comment by EmadAlmahdi** [comment]
I am trying to help the community here .. Is this closed? Or shall I take a look at it?

**Comment by rrsach** [comment]
No, this is still open. I think optimal would be not a config value but a feature where you could pop-out the window on a case-by-case basis.
